### DISCUSSION TITLE
What is the queryOptions in the useSelect() used for?

### DISCUSSION THREAD

**Discussion by BenGardiner123** [refinedev/refine#1656] [Q&A] [answered]
As per the title, it's not entirely clear to me what it does or what it could be useful for?

**Comment by omeraplak** [comment, marked as answer]
Hey @BenGardiner123 ,
Thank you for contact with us. Refine uses [react query](https://react-query.tanstack.com/) to manage state management.

You can use React Query's [options](https://react-query.tanstack.com/reference/useQuery) with queryOptions.

For example,
If you want to make useSelect wait until a condition is met,

```
     const { selectProps: categorySelectProps } = useSelect({
         resource: "categories",
         queryOptions: {
             enabled: someCondition
         }
     });
```

or if you want to use polling,

```
     const { selectProps: categorySelectProps } = useSelect({
         resource: "categories",
         queryOptions: {
             refetchInterval: 1000,
         }
     });
```
